packages/camera/camera_web/lib/src/camera_service.dart

```diff
@@ -220,15 +228,18 @@ class CameraService {
         return null;
       }
 
-      final videoTrackCapabilities = videoTrack.getCapabilities();
+      final Map<dynamic, dynamic> videoTrackCapabilities =
+          videoTrack.getCapabilities();
 
       // A list of facing mode capabilities as
       // the camera may support multiple facing modes.
-      final facingModeCapabilities =
-          List<String>.from(videoTrackCapabilities[_facingModeKey] ?? []);
+      final List<String> facingModeCapabilities = List<String>.from(
+          (videoTrackCapabilities[_facingModeKey] as List<dynamic>?)
+                  ?.cast<String>() ??
```

Comment:
why the explicit .cast<String>() here? can't this be done at the same time as the `as` above?

```dart
videoTrackCapabilities[_facingModeKey] as List<String>?) ?? <String>[]
```

Reply:
Nope. This is an example of what I was talking about in the other comment. `dynamic` is castable to `String`, but `List<dynamic>` isn't castable to `List<String>`.

See https://dartpad.dev/?id=bb6c33398890a6b30c5c5a3db5bdf27f for an example.

Reply:
Ahhh I understand, yikes!

Reply:
This however works in the same Gist:

```dart
  List<Object?> bad = allStrings as List<Object?>; // "as" not even needed!
```

I think `dynamic` can be handled as `Object?` to mitigate what lrhn is commenting [here](https://github.com/dart-lang/language/issues/1324#issuecomment-890493804).

Reply:
Interesting, I'll need to research more. I've definitely been applying the "always use `Object?` instead of `dynamic`" guidance in all other cases, but I've been so wary of things like JSON and method channels after screwing it up several times that I just retreated to "assume that since it's coming in as `dynamic` only `dynamic` is safe", which it sounds like swung too far.